lib/matplotlib/testing/__init__.py

```diff
@@ -136,6 +136,10 @@ def set_font_settings_for_testing():
     rcParams['text.hinting_factor'] = 8
 
 
+def set_reproducibility_for_testing():
+    rcParams['svg.hashsalt'] = 'matplotlib'
```

Comment:
Probably along the same lines as @tacaswell; I was thinking it _wouldn't_ fail if `svg.hashsalt` was broken. However, I see this test doesn't use the decorator anyway, so it is likely to still fail correctly.